### DISCUSSION TITLE
Adding buttons to the title bar of a window?

### DISCUSSION THREAD

**Discussion by DreamSaddle** [emilk/egui#1814] [Q&A]
Hey guys!

Does the `egui` support custom title bar, such as adding buttons to the title bar of a window? If yes, how can I implement it? Thanks for your answer!

For example, something like this.
![image](https://user-images.githubusercontent.com/31559386/178100458-0939c03c-f6af-406d-ae9a-3cd9da3b5d75.png)

**Comment by SacrilegeWasTaken** [comment]
Hello! There is example. I hope it still relevant) 
https://github.com/emilk/egui/blob/master/examples/custom_window_frame/src/main.rs

**Comment by PxlSyl** [comment]
Hi, quite late, but this will be fully supported (like the first picture if you really want it like this) in an updated version of my crate : https://github.com/PxlSyl/egui-desktop

For now, cross platform design, custom menus and keybord navigation are supported, + custom icons. Next version 0.24 will be responsive, and after this (0.25) I plan to add a new api to let you design whatever you want while retaining the native buttons on Mac, Windows, and Linux.
